src/Omnipay/Common/Message/AbstractResponse.php

```diff
@@ -182,8 +183,9 @@ public function getRedirectResponse()
             throw new RuntimeException('This response does not support redirection.');
         }
 
+        /** @var $this RedirectResponseInterface */
         if ('GET' === $this->getRedirectMethod()) {
-            return HttpRedirectResponse::create($this->getRedirectUrl());
+            return new RedirectResponse($this->getRedirectUrl());
```

Comment:
Actually I have a different opinion here. As we are all relying on psr-7 , it would be nice to have / use the proper psr-7 response object than the one provided by 3rd parties. Eg : Zend-Diactoros HtmlResponse , Slim's own etc. That helps not to use those.  I don't know how hard it is to implement. My believe is that it is not that hard though.